### PR TITLE
Add 'switch to instance' buttons next to instances in instance list.

### PR DESCRIPTION
**Pull request by blackle**
screenshot from 2019-02-05 23-57-08

I split `SettingsListItem` into `SettingsListRow` and `SettingsListButton`. The first is responsible for styling its immediate children to look like buttons and have the right padding and borders (so that neighbouring items in the row only have a 1px border between them.) SettingsListButton is responsible for rendering a text button with the special eliding style.

I don't know if this is the best way to do this in terms of structure. It sure makes `_pages/settings/index.html` more verbose...

Maybe I should make a separate component for a row in the instance list and leave the SettingsListItem alone :thinking:

### CODE REVIEW

File: src/routes/_pages/settings/instances/index.html [hunk 1]
```diff
@@ -5,11 +5,16 @@ <h1>Instances</h1>
...
-                        ariaLabel="{instance.name} {instance.name === $currentInstance ? '(current instance)' : ''}" />
+    <SettingsListRow>
+      <SettingsListButton href="/settings/instances/{instance.name}"
+                          label={instance.name}
+                          ariaLabel="{instance.name} {instance.name === $currentInstance ? '(current instance)' : ''}" />
+      <button class="instance-switcher-button" on:click="onSwitchToThisInstance(event, instance.name)">
+        <svg class="instance-switcher-button-svg">
+          <use xlink:href="{instance.name === $currentInstance ? '#fa-star' : '#fa-star-o'}" />
+        </svg>
+      </button>
```
Comment (nolanlawson) [inline src/routes/_pages/settings/instances/index.html]: This should have a "pressed" state and an aria label to indicate what it does. I also like using titles so people can hover and it says what the button does. (For instance, "switch to this instance" and "switch away from this instance".)

**Comment by nolanlawson** [comment]
I love it!! This is such a great idea, I wish I had thought of it myself. :) My only feedback is an accessibility issue, but I can fix that in this branch, no worries.

**Comment by blackle** [comment]
I can fix the issues. I'll do that tomorrow night!

**Comment by nolanlawson** [comment]
No worries, I took care of it! Appreciate it, though. :)

**Comment by nolanlawson** [comment]
BTW please let me know your fediverse handle if you'd like to be mentioned by the @Pinafore@mastodon.technology account when I toot about the next release :smiley:

**Comment by blackle** [comment]
@SuricrasiaOnline@cybre.space

**Pull request merged**